pkg/chunk/grpc/table_client.go

```diff
@@ -0,0 +1,113 @@
+package grpc
+
+import (
+	stdContext "context"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+)
+
+type TableClient struct {
+	cfg        Config
+	client     GrpcStoreClient
+	connection *grpc.ClientConn
+}
+
+// NewStorageClient returns a new StorageClient.
+func NewTableClient(ctx context.Context, cfg Config) (*TableClient, error) {
+	grpcClient, conn, err := connectToGrpcServer(cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	client := &TableClient{
+		cfg:        cfg,
+		client:     grpcClient,
+		connection: conn,
+	}
+	return client, nil
+}
+
+func (c *TableClient) ListTables(ctx stdContext.Context) ([]string, error) {
+	tables, err := c.client.ListTables(ctx, &Nothing{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return tables.Tables, nil
+}
+
+func (c *TableClient) DeleteTable(ctx context.Context, name string) error {
+	tableName := &TableName{TableName: name}
+	_, err := c.client.DeleteTable(ctx, tableName)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	return nil
+}
+
+func (c *TableClient) DescribeTable(ctx context.Context, name string) (desc chunk.TableDesc, isActive bool, err error) {
+	tableName := &TableName{TableName: name}
+	tableDesc, err := c.client.DescribeTable(ctx, tableName)
+	if err != nil {
+		return desc, true, errors.WithStack(err)
```

Comment:
A minor thing, but a good practice when returning errors:

```suggestion
		return desc, false, errors.WithStack(err)
```